### DISCUSSION TITLE
Support for Role-Based Authentication for S3 in Self-Hosted Hive

### DISCUSSION THREAD

**Discussion by bhargavteja07** [graphql-hive/console#5864] [Feature request]
Currently, Hive relies on static IAM user credentials for S3 access, which require additional mechanisms for regular rotation to maintain security. To streamline this and align with AWS best practices, I propose adding support for role-based authentication, allowing for automated, short-lived credentials that enhance security and reduce management overhead.

**Comment by n1ru4l** [comment]
@bhargavteja07 We are happy to review and accept a contribution for this feature request. It is currently, not a priority issue for our hosted solution.
